### DISCUSSION TITLE
How to keep up with upstream django-stubs

### DISCUSSION THREAD

**Discussion by snmishra** [sbdchd/django-types#147] [General]
I like how django-types works with Pyright/Pylance. Thanks for the hard work.

Now I'm facing a dilemma. django-types is far behind django-stubs in completeness and I'm feeling it is falling behind every day. Since the design break is so big, it's hard to port changes added to upstream.

I had hoped that being the default type provider in Pylance will bring more development activity here, but that doesn't seem to be happening.

I'm happy to contribute a few PRs as I find issues, if they're welcome.

**Comment by sbdchd** [comment]
Yeah PRs always welcome!

I really created these types so that auto complete with pylance would work while keeping mypy working for actual type checking in CI

The MO with this repo has been to add things as I need them, rather than following the release notes of Django

I don't really use the latest and greatest that Django has to offer so there hasn't been much changing

**Comment by rchiodo** [comment]
I work on Pylance/Pyright for Microsoft and we use your django-types inside of the shipping code for Pylance. The out of dateness of django-types seems to hit us a lot. 

I was diffing the two, and I was thinking there might be a way to auto merge them?

This might be naive thinking, but it looks like the type stubs that have classes with `_pyi_private` fields in them are using the mypy extension. Every other file seems like it would work well with pyright.

What if there was a CI action that every night pulled changes from django-stubs and merged changes into django-types with the following caveat:
- Any file with `_pyi_private` in it would instead use the existing django-types version

Maybe it wouldn't have to be every night, but say once a week or something.

Thoughts?

**Comment by rchiodo** [comment]
Sorry should have pinged @sbdchd directly. Not sure if you have notifications for replies on or not.

**Comment by rchiodo** [comment]
Prior to the nightly action, I'd submit a PR that attempts to bring things up to date. I think I'd have to manual merge every file, but that seems possible.

**Reply by sbdchd** [reply]
Seems workable! I don't use Django much these days but down to ferry things across
